### PR TITLE
Add Github Actions workflow to execute unit tests on CI

### PR DESCRIPTION
**Pull request by stonko1994**
### Solution

Adding a new GH actions workflow to execute the unit tests on every push and PR.

Pre-work needed:
- Add Fastlane as ruby dependecy
- Add `xcbeautify` through Brew
- Update Quick and Nimble

After the Workflow succeeded I discovered two failing unit tests. Those unit tests were fixed (see comments below for details)

### Checklist
- [x] I added an update to `CHANGELOG.md` file
- [x] I ran all the tests in the project and they succeed

### CODE REVIEW

File: Example/Tests/Helpers/SpyTracker.swift [hunk 1]
```diff
@@ -9,32 +9,44 @@
 import Foundation
 
 class SpyTracker {
-    var spies: [String: [String: String]?] = [:]
+    var spies: [String: [[String: String]?]] = [:]
```
Comment (stonko1994) [inline Example/Tests/Helpers/SpyTracker.swift:12]: Previously this was only holding the last call to a function. In the case where a spy is called multiple times, previous calls were lost. I extended it to keep all calls to that expectations can also verify previous calls.

File: Example/Tests/Helpers/SpyTracker.swift [hunk 2]
```diff
@@ -9,32 +9,44 @@
...
+            return (false, [])
         }
 
         if let expectedArgs = withArgs {
-            if let calledArgs = spies[name] {
-                var containsExpectedArgs = true
-                for key in expectedArgs.keys {
-                    containsExpectedArgs = containsExpectedArgs && (calledArgs?[key] == expectedArgs[key])
+            if let spyCalls = spies[name] {
+                for calledArgs in spyCalls {
```
Comment (stonko1994) [inline Example/Tests/Helpers/SpyTracker.swift:31]: Please keep in mind that I did not do quality refactoring here when reviewing.

File: Example/Tests/CISVideoAnalyticsTestDouble.swift [hunk 3]
```diff
@@ -39,6 +39,7 @@ class CISVideoAnalyticsTestDouble: NSObject, CISVideoAnalyticsProtocol, TestDoub
             "streamUrl": "\(contentInfo?[CIS_SSDK_METADATA_STREAM_URL] ?? "")",
             "viewerId": "\(contentInfo?[CIS_SSDK_METADATA_VIEWER_ID] ?? "")",
             "defaultResource": "\(contentInfo?[CIS_SSDK_METADATA_DEFAULT_RESOURCE] ?? "")",
+            "encodedFrameRate": "\(contentInfo?[CIS_SSDK_METADATA_ENCODED_FRAMERATE] ?? "")",
```
Comment (stonko1994) [inline Example/Tests/CISVideoAnalyticsTestDouble.swift:42]: This was missing although a test was expecting it.

File: Example/Tests/Helpers/SpyTracker.swift [hunk 4]
```diff
@@ -9,32 +9,44 @@
...
-    var spies: [String: [String: String]?] = [:]
+    var spies: [String: [[String: String]?]] = [:]
 
     func track(functionName: String, args: [String: String]? = nil) {
-        spies[functionName] = args
+        if spies[functionName] == nil {
+            spies[functionName] = []
+        }
+
+        spies[functionName]?.append(args)
```
Comment (rolandkakonyi) [inline Example/Tests/Helpers/SpyTracker.swift:19]: nit: you could do this instead to make it simpler.

```suggestion
        spies[functionName, default: []].append(args)
```

**Review by rolandkakonyi — APPROVED** [review]

**Pull request merged**